web/static/js/components/thesis-editor.js

```diff
@@ -53,50 +46,39 @@ class ThesisEditor extends React.Component {
     this.addPagePressed = this.addPagePressed.bind(this)
     this.deletePagePressed = this.deletePagePressed.bind(this)
     this.pageSettingsPressed = this.pageSettingsPressed.bind(this)
-  }
 
-  warnURLRedirect (url) {
-    if (!url) return
-    if (window.confirm(`This page is set to redirect to ${url}. Follow redirect?`)) {
-      window.location = url
-    }
-  }
-
-  ospryPublicKey () {
-    return thesisContainer.getAttribute('data-ospry-public-key')
-  }
-
-  pathname () {
-    return window.location.pathname
-  }
-
-  pageTitle () {
-    return document.title
-  }
-
-  pageDescription () {
-    const desc = this.pageDescriptionMetaTag()
-    return desc ? desc.content : null
-  }
+    // External editors
+    
+    this.htmlEditor = new HtmlEditor({
+      onChange: () => this.setState({pageModified: true})
+    })
```

Comment:
External editors are now initialized with callback function(s) that allow them to affect the parent component without having a direct reference, which was icky feeling.